open_dread_rando/lua_editor.py

```diff
@@ -76,8 +98,12 @@ def get_script_class(self, pickup_resources: list[dict], boss: bool = False) ->
     def add_progressive_class(self, replacement):
         new_class = lua_util.replace_lua_template("randomizer_progressive_template.lua", replacement)
         self._powerup_script += new_class.encode("utf-8")
+    
+    def add_progressive_models(self, replacement):
```

Comment:
I'd say that this function should do more of the work of knowing what arguments `randomizer_progressive_template.lua`.
Basically, move a good chunk of the code that's in `get_script_class` right now to here.